encoding/codecv2/codecv2.go

```diff
@@ -0,0 +1,618 @@
+package codecv2
+
+/*
+#cgo LDFLAGS: -L../../rs -lm -ldl -lscroll_zstd
+#include <stdlib.h>
+#include "../../rs/rs_zstd.h"
+*/
+import "C"
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+	"strings"
+	"unsafe"
+
+	"github.com/scroll-tech/go-ethereum/accounts/abi"
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/core/types"
+	"github.com/scroll-tech/go-ethereum/crypto"
+	"github.com/scroll-tech/go-ethereum/crypto/kzg4844"
+	"github.com/scroll-tech/go-ethereum/log"
+
+	"github.com/scroll-tech/da-codec/encoding"
+)
+
+var (
+	// BLSModulus is the BLS modulus defined in EIP-4844.
+	BLSModulus *big.Int
+
+	// BlobDataProofArgs defines the argument types for `_blobDataProof` in `finalizeBatchWithProof4844`.
+	BlobDataProofArgs abi.Arguments
+
+	// MaxNumChunks is the maximum number of chunks that a batch can contain.
+	MaxNumChunks int = 45
+)
+
+func init() {
```

Comment:
change `init` to `Newxxx() error`, outside can get the error.

Reply:
`init()` and `log.Crit()` are removed in [1f826c0](https://github.com/scroll-tech/da-codec/pull/2/commits/1f826c00dd47706c55a4cf6b580c01a3bc05f671).